models/user/user.go

```diff
@@ -814,8 +814,8 @@ func validateUser(u *User) error {
 }
 
 // UpdateUser updates user's information.
-func UpdateUser(ctx context.Context, u *User, changePrimaryEmail bool, cols ...string) error {
-	err := validateUser(u)
+func UpdateUser(ctx context.Context, u *User, changePrimaryEmail, visibilityChanged bool, cols ...string) error {
+	err := validateUser(u, visibilityChanged)
```

Comment:
It confuses me, why `visibilityChanged` means `checkVisibility`? Even if it is correct, it's hard to understand.

As a quick patch, this PR might work. 

But I think it reduces maintainability,  I can't imagine what does `UpdateUser(ctx, u, false, false)` or `UpdateUser(ctx, u, false, true)` do without reading all related code.

Reply:
How about add an error handling after calling `validateUser` instead of add `checkVisibility` to `validateUser`?

``` go
if err := validateUser(u); err != nil {
  if !(ignoreVisibilityNotAllowed && IsVisibilityNotAllowed(err)) {
    return err
  }
}
```

Reply:
> It confuses me, why `visibilityChanged` means `checkVisibility`? Even if it is correct, it's hard to understand.
> 
> As a quick patch, this PR might work.
> 
> But I think it reduces maintainability, I can't imagine what does `UpdateUser(ctx, u, false, false)` or `UpdateUser(ctx, u, false, true)` do without reading all related code.

That means the previous value which have stored in database should not be checked, only changed values need to be checked .

Reply:
I can understand after I spend much time on reading code, while I do not think it is a good design.

Now we have `(..., true, false)`, do we want to have `(..., true, false, true)` in the future?